packages/autorest.python/autorest/preprocess/__init__.py

```diff
@@ -171,6 +171,28 @@ def headers_convert(yaml_data: Dict[str, Any], replace_data: Any) -> None:
             yaml_data[k] = v
 
 
+def _has_special_content_type(
```

Comment:
this code is getting way too complicated, can you simplify this whole process, thanks

Reply:
Simplified. I also added some annotations